gui/default/syncthing/device/editDeviceModalView.html

```diff
@@ -10,19 +10,16 @@
         <div id="device-general" class="tab-pane in active">
           <div ng-if="!editingDeviceDefaults()" class="form-group" ng-class="{'has-error': deviceEditor.deviceID.$invalid && deviceEditor.deviceID.$dirty}" ng-init="loadFormIntoScope(deviceEditor)">
             <label translate for="deviceID">Device ID</label>
-            <div ng-if="!editingDeviceExisting()">
-              <div class="input-group">
-                <input name="deviceID" id="deviceID" class="form-control text-monospace" type="text" ng-model="currentDevice.deviceID" required="" valid-deviceid list="discovery-list" aria-required="true" />
-                <div class="input-group-btn">
-                  <button type="button" class="btn btn-default" data-toggle="modal" data-target="#idqr" ng-disabled="!deviceEditor.deviceID.$valid">
-                    <span class="fas fa-qrcode"></span>&nbsp;<span translate>Show QR</span>
-                  </button>
-                </div>
+            <div class="input-group">
+              <input name="deviceID" id="deviceID" class="form-control text-monospace" type="text" ng-model="currentDevice.deviceID" required="" valid-deviceid list="discovery-list" aria-required="true" ng-readonly="currentDevice._editing !== 'add'" />
+              <div class="input-group-btn">
+                <button type="button" class="btn btn-default" data-toggle="modal" data-target="#idqr" ng-disabled="!deviceEditor.deviceID.$valid">
+                  <span class="fas fa-qrcode"></span>&nbsp;<span translate>Show QR</span>
+                </button>
               </div>
-              <datalist id="discovery-list">
-                <option ng-repeat="id in discovery" value="{{id}}" />
-              </datalist>
-              <p class="help-block" ng-if="discovery && discovery.length !== 0">
+            </div>
+            <div ng-if="currentDevice._editing === 'add'">
```

Comment:
> And please use equal (`==`) checking instead of identity (`===`) for strings.

Fixed.

I haven't done anything with the internal variables though. Is it absolutely necessary? The `currentDevice._editing` is already present in the currently existing code (see https://github.com/tomasz1986/syncthing/blob/8d5f68c857fe0d85a0db349b52222ea8b3ffbd30/gui/default/syncthing/device/editDeviceModalView.html#L168), so I've just tried to follow a similar convention here.